source/posts/2020-09-10-cross-browser-testing-with-non-functional-tests.html.markdown.erb

```diff
@@ -0,0 +1,194 @@
+---
+layout: post
+post_title: Crossbrowser testing with Non-functional tests
+date: 2020-09-10
+summary_image: "/assets/images/blog/crossbrowser_testing_with_non_functional_tests.jpg"
+excerpt: "Here's why functional tests are not meant for cross browser testing"
+author_name: "Soumya Swaroop"
+author_image: "https://cdn-images-1.medium.com/fit/c/100/100/1*P0xkevj9Mngmm7NVAp5KIQ.jpeg"
+published: true
+title: "Crossbrowser testing with Non-functional tests | Gauge Blog"
+title_tag_of_header: "Crossbrowser testing with Non-functional tests | Gauge Blog"
+meta_description: "Here's why functional tests are not meant for cross browser testing"
+meta_keywords: "taiko, test automation, cross browser, static code analysis"
+---
+
+#### Sep 10, 2020 | Soumya Swaroop
+
+<%= image_tag "blog/crossbrowser_testing_with_non_functional_tests_large.jpg", {:title => "Crossbrowser testing with Non-functional tests", :alt => "Crossbrowser testing with Non-functional tests"} %>
+
+# Cross browser testing with non-functional tests
+
+Ensuring [cross browser compatibility](https://en.wikipedia.org/wiki/Cross-browser_compatibility) 
+is a [non functional requirement](https://en.wikipedia.org/wiki/Non-functional_requirement) 
+(NFR). Cross browser requirements do not define the system's behavior but 
+define criteria where the system will be used.
+
+> A [functional requirement](https://en.wikipedia.org/wiki/Functional_requirement) 
+defines what a system is supposed to do. A NFR specifies criteria that can be used 
+to judge the operation of a system, rather than specific behaviors. 
+- [wikipedia](https://en.wikipedia.org/wiki/Non-functional_requirement).
+
+In my roles as a developer and tester, I see teams only using automated 
+UI tests, originally written to test the functionality, run across browsers to 
+ensure compatibility. Here, functional tests are used as a blanket to cover 
+the cross browser issues. 
+
+However, after working on [Gauge](https://gauge.org/) and [Taiko](https://docs.taiko.dev/), 
+I feel there’s a better and faster way to check cross browser issues than 
+relying only on functional tests. 
+
+In this post we look at ways to ensure cross browser compatibility without 
+relying on functional tests. Not all cross browser compatibility issues are 
+caught by functional tests. Here, I want to highlight how teams are using 
+alternatives to address them.
```

Comment:
Change whole paragraph to

Not all cross browser compatibility issues are caught by functional tests. 
In this post we look at ways to ensure cross browser compatibility without 
relying on functional tests.